open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/graph-repository-connector/src/main/java/org/odpi/openmetadata/adapters/repositoryservices/graphrepository/repositoryconnector/GraphOMRSMetadataStore.java

```diff
@@ -574,7 +574,7 @@ synchronized EntityDetail getEntityDetailFromStore(String guid)
         {
 
             // Entity was not found by GUID
-            log.error("{} entity with GUID {} not found", methodName, guid);
+            log.warn("{} entity with GUID {} not found", methodName, guid);
```

Comment:
this method is being called by methods like updateEntity & classifyEntity. Maybe it is reasonable if you attempt to classify or update an entity that does not exist then this is logged as an error. After all it is resulting in a transaction rollback.

I am making these comments as the pr issue description seems to be talking about get, but the fix is around getting for update or the like. I assume there is an logic error in one of the calling layers, if we attempt to update (etc) an entity that does not exist.

Reply:
> I am making these comments as the pr issue description seems to be talking about get, but the fix is around getting for update or the like. I assume there is an logic error in one of the calling layers, if we attempt to update (etc) an entity that does not exist.

Maybe the description does not completely ilustrate the reasoning behind this change. The problem should not be observed for simple use-cases like get single instance, check if instance exists or classify/update entity (in your examples) but combined sequence of calls in a process like initial load or create process lineage using Data Engine OMAS. 

It is not about the error handling, that is properly implemented in the upper layers. Depending on the operation, exceptions are propagated and logged using auditlog.

This change only addresses developer/debug information that is outputted when we are trying to test or troubleshot some iregular behaviour. This problem becomes evident in the case when you have more compex deployment with mutlipe repositories (native + proxies)... then you want to have the ability to determine what level of details you want to see as developer. From experiance, we see that most of the time 'not found' messages are not actual error but regular logical events that are handled as part of more complex logic (exampe is isEntityKnow will now log error because there is entityProxy present which in some cases with external repository is just normal)

Reply:
@lpalashevski  Sounds fine - maybe we should put this principle in the developers guide - with this example.